src/components/single/SingleArticlePage.jsx

```diff
@@ -43,21 +46,55 @@ export const SingleArticlePage = () => {
                 username: newComment.author,
                 body: newComment.body,
             })
+            .then(posted => {
+                setCreatedCommentId(posted.data.comment.comment_id)
+            })
             .catch(err => {
-                setComments((currComments) => [currComments.shift(), ...currComments])
+                setComments((currComments) => {
+                    return [...currComments].filter(comment => comment.comment_id !== undefined)
+                })
                 console.log(err)
-                toast.error("Your comment hasn't been published!")
+                toast.error("Your comment hasn't been published!", {
+                    toastId: "error"        
+                })
             })
         }
     }, [newComment])
 
+    useEffect(() => {
+        setComments((currComments) => {
+            return [...currComments].filter(comment => {
+                if (comment.comment_id == undefined) {
+                    comment.comment_id = createdCommentId
+                } else {
+                    return comment
+                }
+            })
+        })
+    }, [createdCommentId])
```

Comment:
Same as the other useEffect this one isn't needed either. You can do this logic directly instead of setting `createdCommentId` in state